pycbc/events/significance.py

```diff
@@ -0,0 +1,318 @@
+
+    # If the foreground are *quieter* than the background or at the same value
+    # then the search sorted algorithm will choose position -1, which does not
+    # exist. We force it back to zero.
+    if isinstance(idx, np.ndarray):  # Case where our input is an array
+        idx[idx < 0] = 0
+    else:  # Case where our input is just a scalar value
+        if idx < 0:
+            idx = 0
+
+    fore_n_louder = n_louder[idx]
+
+    if not skip_background:
+        unsort = sort.argsort()
+        back_cum_num = n_louder[unsort]
+        return back_cum_num, fore_n_louder
+
+    return fore_n_louder
+
+
+def n_louder_from_fit(back_stat, fore_stat, dec_facs,
+                      fit_func='exponential', fit_thresh=0):
+    """
+    Use a fit to events in back_stat in order to estimate the
+    distribution for use in recovering the estimate count of louder
+    background events. Below the fit threshold, use the n_louder
+    method for these triggers
+
+    back_stat: numpy.ndarray
+        Array of the background statistic values
+    fore_stat: numpy.ndarray or scalar
+        Array of the foreground statistic values or single value
+    dec_facs: numpy.ndarray
+        Array of the decimation factors for the background statistics
+    fit_func: str
+        Name of the function to be used for the fit to background
+        statistic values
+    fit_thresh: float
+        Threshold above which triggers use the fitted value, below this
+        the counted number of louder events will be used
+
+    Returns
+    -------
+    back_cnum: numpy.ndarray
+        The estimated number of background events louder than each
+        background event
+    fn_louder: numpy.ndarray
+        The estimated number of background events louder than each
+        foreground event
+    """
+
+    # Calculate the fitting factor of the ranking statistic distribution
+    alpha, _ = trstats.fit_above_thresh(fit_func, back_stat,
+                                        thresh=fit_thresh,
+                                        weights=dec_facs)
+
+    # Count background events above threshold as the cum_fit is
+    # normalised to 1
+    bg_above = back_stat > fit_thresh
+    bg_above_thresh = np.sum(dec_facs[bg_above])
+    fg_above = fore_stat > fit_thresh
+
+    # These will be overwritten, but just to silence a warning
+    # in the case where trstats.cum_fit returns zero
+    back_cnum = np.zeros_like(back_stat)
+    fnlouder = np.zeros_like(fore_stat)
+
+    # Ue the fit above the threshold
+    back_cnum[bg_above] = trstats.cum_fit(fit_func, back_stat[bg_above],
+                                          alpha, fit_thresh) * bg_above_thresh
+    fnlouder[fg_above] = trstats.cum_fit(fit_func, fore_stat[fg_above],
+                                         alpha, fit_thresh) * bg_above_thresh
+
+    # below the fit threshold, we count the number of louder events,
+    # as things get complicated by clustering below this point
```

Comment:
The comment may or may not be true depending on the data and the choice of threshold .. it seems to be explaining motivation. 
Suggest eg 'Below the fit threshold we expect there to be sufficient events to use the count louder method, and the distribution may deviate from the fit function '